src/tools/compiletest/src/header.rs

```diff
@@ -925,6 +925,7 @@ const KNOWN_DIRECTIVE_NAMES: &[&str] = &[
     "test-mir-pass",
     "unset-exec-env",
     "unset-rustc-env",
+    ("unused-revision-names"/* used by tidy check `unknown_revision` */),
```

Comment:
These parentheses are here because if I just add a normal end-of-line comment, `rustfmt` insists on reindenting the `// tidy-alphabetical-end` to be aligned with it.

I'm open to other suggestions.

Reply:
Wouldn't this work?
```suggestion
    // used by tidy check `unknown_revision`
    "unused-revision-names",
```

Reply:
Oh, you might be right. I had mentally ruled it out for reasons that aren’t relevant in this context.

Reply:
Also what I have in mind for compiletest directive parsing is that if we rework the parsing to stop silently allowing unknown directives to go through, we wouldn't need this hacky `KNOWN_DIRECTIVE_NAMES` array which isn't exhaustive anyway and requires people to modify it every time a directive is added or removed or renamed.